tests/tt_metal/tt_metal/module.mk

```diff
@@ -72,7 +72,7 @@ TT_METAL_TESTS += \
 TT_METAL_TESTS_SRCS = $(addprefix tests/tt_metal/, $(addsuffix .cpp, $(TT_METAL_TESTS:tests/%=%)))
 
 TT_METAL_TESTS_INCLUDES = $(TEST_INCLUDES) $(TT_METAL_INCLUDES)
-TT_METAL_TESTS_LDFLAGS = -ltt_metal -ldl -lstdc++fs -pthread -lyaml-cpp -lgtest
+TT_METAL_TESTS_LDFLAGS = -ltt_metal -ldl -lstdc++fs -pthread -lyaml-cpp -lgtest -lm
```

Comment:
math lib?

Reply:
This came in from Nick when he enabled LLVM building. My guess is that with GCC you implicitly got it included through dependencies and LLVM was pickier. We might have more minor items like this trickling in after this main set.

Reply:
How priority of an item is it to start running on more distros / toolchains? We have an open to create a job on Github hosted runners to start compiling and uploading metal build artifacts on more build configs and platforms: https://github.com/tenstorrent-metal/tt-metal/issues/4416

Probably going to matrix over a couple of Ubuntus, and can include other toolchains. More LLVM is also in my scope. Other archs like ARM and RISC-V... lowkey dicey but still open to it. If we could get access to some internally that could work nicely, although that would technically be a self-hosted runner. I don't believe many cloud providers, if any, have generally-available RISC-V hosts.

We could specify which we want and throw those into that issue too as a comment.

Reply:
I had hoped to start chipping away at this, and do it best-effort style since I want it myself. Once we've done the initial cleanup (such as this patch series), we should probably setup at least build jobs for these toolchains. Happy to keep chipping away at it.

External feedback from GS owners is that they'd like to move away from the python 3.8, etc, restrictions.